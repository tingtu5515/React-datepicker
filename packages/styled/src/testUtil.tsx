import React from 'react'
import {render} from '@testing-library/react'
import DatepickerContext, {datepickerContextDefaultValue} from './context/datepickerContext'

function renderWithProviders(ui: any, options?: any, contextValues?: Record<string, unknown>) {
  return render(
    <DatepickerContext.Provider value={{...datepickerContextDefaultValue, ...contextValues}}>
      {ui}
    </DatepickerContext.Provider>,
    options,
  )
}

export * from '@testing-library/react'
export {renderWithProviders as render}
