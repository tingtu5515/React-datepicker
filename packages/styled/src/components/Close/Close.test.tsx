import React from 'react'
import {render, fireEvent} from '../../testUtil'
import CloseModal from '.'

test('should match snapshot', () => {
  const {container} = render(<CloseModal rtl={false} onClick={jest.fn()} />)
  expect(container).toMatchSnapshot()
})

test('should render rtl variant', () => {
  const {container} = render(<CloseModal rtl={false} onClick={jest.fn()} />)
  expect(container).toMatchSnapshot()
})

test('should execute callback', () => {
  const onClick = jest.fn()
  const {container} = render(<CloseModal rtl={false} onClick={onClick} />)
  // @ts-ignore
  fireEvent.click(container.firstChild)
  expect(onClick).toHaveBeenCalled()
})
