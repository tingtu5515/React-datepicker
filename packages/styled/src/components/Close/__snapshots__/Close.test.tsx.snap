// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
.c2 {
  margin: 1px 0 0 16px;
  color: #929598;
  font-size: 12px;
  font-family: Montserrat,sans-serif;
  font-weight: 600;
  font-size: 12px;
  font-family: Montserrat,sans-serif;
  font-weight: 600;
  font-size: 12px;
  font-family: Montserrat,sans-serif;
  font-weight: 600;
  float: left;
  -webkit-transition: color 0.15s;
  transition: color 0.15s;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  background: transparent;
  padding: 0;
  border: 0;
}

.c0 svg {
  -webkit-transition: color 0.15s;
  transition: color 0.15s;
}

.c0:hover .c1 {
  color: #343132;
}

.c0:hover svg {
  color: #343132;
}

<div>
  <button
    aria-label="Close"
    class="c0"
    color="#343132"
    data-testid="DatepickerClose"
    tabindex="-1"
  >
    <svg
      class=""
      color="#ADADAD"
      height="16px"
      viewBox="0 0 15 16"
      width="15px"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M14.69.263a.802.802 0 0 0-1.187 0L7.47 6.694 1.433.262a.802.802 0 0 0-1.187 0 .938.938 0 0 0 0 1.267L6.28 7.96.246 14.392a.937.937 0 0 0 0 1.266.81.81 0 0 0 .594.262.81.81 0 0 0 .593-.262l6.035-6.432 6.035 6.432a.812.812 0 0 0 .593.262.81.81 0 0 0 .594-.262.937.937 0 0 0 0-1.266L8.656 7.96l6.034-6.43a.937.937 0 0 0 0-1.267z"
        fill="currentColor"
        fill-rule="nonzero"
      />
    </svg>
    <div
      class="c1 c2"
      color="#929598"
      font-family="Montserrat, sans-serif"
      font-size="12px"
      font-weight="600"
    >
      Close
    </div>
  </button>
</div>
`;

exports[`should render rtl variant 1`] = `
.c2 {
  margin: 1px 0 0 16px;
  color: #929598;
  font-size: 12px;
  font-family: Montserrat,sans-serif;
  font-weight: 600;
  font-size: 12px;
  font-family: Montserrat,sans-serif;
  font-weight: 600;
  font-size: 12px;
  font-family: Montserrat,sans-serif;
  font-weight: 600;
  float: left;
  -webkit-transition: color 0.15s;
  transition: color 0.15s;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  background: transparent;
  padding: 0;
  border: 0;
}

.c0 svg {
  -webkit-transition: color 0.15s;
  transition: color 0.15s;
}

.c0:hover .c1 {
  color: #343132;
}

.c0:hover svg {
  color: #343132;
}

<div>
  <button
    aria-label="Close"
    class="c0"
    color="#343132"
    data-testid="DatepickerClose"
    tabindex="-1"
  >
    <svg
      class=""
      color="#ADADAD"
      height="16px"
      viewBox="0 0 15 16"
      width="15px"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M14.69.263a.802.802 0 0 0-1.187 0L7.47 6.694 1.433.262a.802.802 0 0 0-1.187 0 .938.938 0 0 0 0 1.267L6.28 7.96.246 14.392a.937.937 0 0 0 0 1.266.81.81 0 0 0 .594.262.81.81 0 0 0 .593-.262l6.035-6.432 6.035 6.432a.812.812 0 0 0 .593.262.81.81 0 0 0 .594-.262.937.937 0 0 0 0-1.266L8.656 7.96l6.034-6.43a.937.937 0 0 0 0-1.267z"
        fill="currentColor"
        fill-rule="nonzero"
      />
    </svg>
    <div
      class="c1 c2"
      color="#929598"
      font-family="Montserrat, sans-serif"
      font-size="12px"
      font-weight="600"
    >
      Close
    </div>
  </button>
</div>
`;
